### ISSUE TITLE
MDX/addon-docs bug: Cannot use <Story>

### ISSUE THREAD

**Issue by eirslett** [eirslett/storybook-builder-vite#40]
Steps to reproduce: Tweak the example-react project, add

```
import { Meta, Canvas, Story } from '@storybook/addon-docs';
```

to Introduction.stories.mdx, and then

```
 # Welcome to Storybook

<Canvas>
    <Story name="testStory">
        <div>Oh no it doesn't work</div>
    </Story>
</Canvas>
```

We get:

```
@storybook_addon-docs.js?v=f78cd456:10690 Uncaught TypeError: Cannot read property 'fromId' of undefined
    at getStoryProps2 (@storybook_addon-docs.js?v=f78cd456:10690)
```

The error comes from addon-docs/blocks/Story.tsx (the Story component), inside `getStoryProps` where `storyStore` for some reason is undefined.

**Comment by eirslett** [comment]
@shilman I think you're familiar with that code? Looks like storyStore is supposed to come from a React Context somehow? There's a missing null/undefined check in TypeScript in `getStoryProps`, and I tried patching that - Storybook runs, but unfortunately the source code is not loaded into the source code preview component.

**Comment by julrich** [comment]
This seems to be related to / a duplicate of https://github.com/eirslett/storybook-builder-vite/issues/13, I think?
At least the `fromId` exception is exactly what we're hitting!

**Comment by eirslett** [comment]
That's right, looks like it's the same bug! For some reason the storyStore is not set up correctly with the Vite builder.

**Comment by shilman** [comment]
This is where it should be getting set up:

**Comment by eirslett** [comment]
In `start.ts`:

```
function getClientApi(decorateStory: DecorateStoryFunction, channel?: Channel) {
  let storyStore: StoryStore;
  let clientApi: ClientApi;
  if (
    typeof globalWindow !== 'undefined' &&
    globalWindow.__STORYBOOK_CLIENT_API__ &&
    globalWindow.__STORYBOOK_STORY_STORE__
  ) {
    clientApi = globalWindow.__STORYBOOK_CLIENT_API__;
    storyStore = globalWindow.__STORYBOOK_STORY_STORE__;
  } else {
    storyStore = new StoryStore({ channel });
    clientApi = new ClientApi({ storyStore, decorateStory });
  }
  return { clientApi, storyStore };
}
```

Could be something with the `globalWindow` thing that is not working like expected, so that it's not picking up `storyStore` from `window`?

**Comment by eirslett** [comment]
I think I've traced this down to a Vite/esbuild dev issue - because in my testing, it works as expected with the production/static Storybook build. (Vite uses rollup instead of esbuild for production)

For some reason Vite is loading the `DocsContext.ts` file twice, which ruins the identity of the context variable - and then we lose the context that React is supposed to pass through the component tree.

**Comment by IanVS** [comment]
@eirslett what version of vite are you using?  I had problems with loading contexts in different ways in 4.2.0-beta.0 (https://github.com/vitejs/vite/issues/4005), I wonder if something similar is happening to you.

**Comment by eirslett** [comment]
Looks like I'm using 2.3.7, I can reproduce it in this repo. What I did to debug it was to manually edit the file inside `node_modules` that defines the React context (in this case `DocsContext` and `DocsContext.ts` from Storybook), and then `console.log("Hello world, this is from within the file that defines the React context");` and in the console log, I would get that entry two times, from two different sources. Is that what you experience too?

(When you make changes to files inside `node_modules`, you first have to stop Vite, then delete the `node_modules/.vite` folder, restart, and then force-refresh the browser.)

**Comment by eirslett** [comment]
This should be fixed in the latest release (with Storybook 6.3.4 or newer)
